File: patterns/c-patterns/pattern129.c
// Pattern 129: X-Max Tree

//         *
//        * *
//       * * *
//      * * * *
//       * * *
//      * * * *
//     * * * * *
//    * * * * * *
//      * * * *
//      * * * *
//      * * * *
//      * * * *

int main()
{
    int i, j, k;
    int height = 3, width = 4;

    int space = width * 5;
    int r = 1;
    int m = 1;

    for (r = 1; r < height; r++)
    {
        for (i = m; i <= width; i++)
        {
            for (j = space; j >= i; j--)
            {
                printf(" ");
            }
            for (k = 1; k <= i; k++)
            {
                printf("* ");
            }
            printf("\n");
        }
        m += 2;
        width += 2;
    }
    for (i = 1; i <= 4; i++)
    {
        for (j = space - 3; j >= 1; j--)
        {
            printf(" ");
        }
        for (k = 1; k <= 4; k++)
        {
            printf("* ");
        }
        printf("\n");
    }
}

File: patterns/python-patterns/pattern129.py
# Pattern 129: X-Max Tree

#         *
#        * *
#       * * *
#      * * * *
#       * * *
#      * * * *
#     * * * * *
#    * * * * * *
#      * * * *
#      * * * *
#      * * * *
#      * * * *

height = 3
width = 4

space = width * 5
r = 1
m = 1

for r in range(1, height):
    for i in range(m, width + 1):
        for j in range(space, i - 1, -1):
            print(" ", end="")
        for k in range(1, i + 1):
            print("* ", end="")
        print()
    m += 2
    width += 2

for i in range(1, 5):
    for j in range(space - 3, 0, -1):
        print(" ", end="")
    for k in range(1, 5):
        print("* ", end="")
    print()
